ui/src/app/views/notebook-list/component.ts

```diff
@@ -158,6 +166,58 @@ export class NotebookListComponent implements OnInit, OnDestroy {
     this.openNotebook();
   }
 
+  renameThis(notebook: FileDetail): void {
+    this.notebookInFocus = notebook;
+    this.renameModal.open();
+  }
+
+  receiveRename($event): void {
+    this.renameNotebook($event);
+  }
+
+  renameNotebook(rename): void {
+    let newName = rename.newName;
+    if (!(new RegExp('^.+\.ipynb$').test(newName))) {
```

Comment:
We should have some protection against `/`s in the notebook name - these will not work as expected since "list-notebooks" doesn't glob into subdirectories.